usim/_primitives/condition.py

```diff
@@ -97,8 +98,13 @@ def __init__(self, *conditions: Condition):
         # unpack similar connections
         # eliminate duplicates
         self._children = tuple(
-            set(condition for condition in conditions if not isinstance(condition, self.__class__)) |
-            set(condition._children for condition in conditions if isinstance(condition, self.__class__))
+            set(condition for condition in conditions if not isinstance(
+                condition, self.__class__
+            )).union(
+                condition._children for condition in conditions if isinstance(
+                    condition, self.__class__
+                )
+            )
```

Comment:
This is very bad maintainable. We should consider for refactoring.

Reply:
This actually has a bug as well. :D See issue #5, I'll open a pull request once the CI is in master.